api/api.go

```diff
@@ -593,7 +607,28 @@ func (api *API) listSilences(w http.ResponseWriter, r *http.Request) {
 		sils = append(sils, s)
 	}
 
-	respond(w, sils)
+	var active, pending, expired, silences []*types.Silence
+
+	for _, s := range sils {
+		switch s.Status.State {
+		case "active":
+			active = append(active, s)
+		case "pending":
+			pending = append(pending, s)
+		case "expired":
+			expired = append(expired, s)
+		}
+	}
+
+	sort.Sort(ByEndTime(active))
+	sort.Sort(ByStartTime(pending))
```

Comment:
These could be replaced with https://golang.org/pkg/sort/#Slice

Reply:
I've replaced with your suggestion 👍